src/librustc/mir/mod.rs

```diff
@@ -1500,6 +1501,13 @@ impl<'tcx> BasicBlockData<'tcx> {
         self.terminator.as_mut().expect("invalid terminator state")
     }
 
+    pub fn is_unreachable(&self) -> bool {
```

Comment:
IMO this should check for no statements, and probably be inlined in its only callsite, with a FIXME comment, since IIRC we should be able to get rid of this check at some point?

Reply:
I'll do that, but I don't recall a reason we'd be able to remove this.

Reply:
r=me with this `pub fn is_unreachable` removed.

Reply:
Sigh, one of those days..